src/Mvc/Mvc.Core/src/Microsoft.AspNetCore.Mvc.Core.csproj

```diff
@@ -27,6 +27,10 @@ Microsoft.AspNetCore.Mvc.RouteAttribute</Description>
     <Compile Include="$(SharedSourceRoot)SecurityHelper\**\*.cs" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Compile Include="..\..\..\Http\Routing\src\DataSourceDependentCache.cs" Link="Routing\DataSourceDependentCache.cs" />
```

Comment:
Move the file to a Shared folder?

Reply:
*eh* - I don't really free like this helps.

Reply:
I don't like a project in a different solution referencing a file. It will be easy for someone to change the API and only discover another solution is referencing it when the build fails.

Reply:
I don't really like it either. I left a comment in `DataSourceDependentCache` so that it's more obvious. I don't see how moving the file on disk helps.